Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -1251,28 +1249,22 @@ void Device::set_light_model_params(const LightModelParameters& lighting_model)
 ColorType Device::get_color_buffer_pixel(int x, int y)
 {
     // FIXME: Reading individual pixels is very slow, rewrite this to transfer whole blocks
-    if (x < 0 || y < 0 || x >= m_frame_buffer->rect().width() || y >= m_frame_buffer->rect().height())
+    if (!m_frame_buffer->rect().contains(x, y))
         return 0;
-
     return m_frame_buffer->color_buffer()->scanline(y)[x];
 }
 
 DepthType Device::get_depthbuffer_value(int x, int y)
 {
     // FIXME: Reading individual pixels is very slow, rewrite this to transfer whole blocks
-    if (x < 0 || y < 0 || x >= m_frame_buffer->rect().width() || y >= m_frame_buffer->rect().height())
+    if (!m_frame_buffer->rect().contains(x, y))
         return 1.0f;
-
     return m_frame_buffer->depth_buffer()->scanline(y)[x];
 }
 
 NonnullRefPtr<Image> Device::create_image(unsigned width, unsigned height, unsigned depth, unsigned levels, unsigned layers)
 {
-    VERIFY(width > 0);
-    VERIFY(height > 0);
-    VERIFY(depth > 0);
-    VERIFY(levels > 0);
-    VERIFY(layers > 0);
+    VERIFY(width > 0 && height > 0 && depth > 0 && levels > 0 && layers > 0);
```

Comment:
Isn't it better to have these as separate VERIFYs? Then we would actually know which of the arguments was zero on failure.

Reply:
I originally thought VERIFY had some more behind the scenes macro magic going on, but it's a simple `assert` - so I reverted this change. Thanks!